recipe/run_test.sh

```diff
@@ -0,0 +1,8 @@
+# Skip compile tests if we are cross-compiling
+if [[ "${build_platform}" != "${target_platform}" ]]; then
+  exit 0
+fi
+
+# Test boost::iostreams zlib filter support
+cd test
+${CXX} -I$PREFIX/include -L$PREFIX/lib test_iostreams_zlib.cpp -lboost_iostreams
```

Comment:
Forget whether we need to run these scripts ourselves when using `test/commands` or not. Did these get run in the previous change? It's not clear from the logs.

Reply:
Yes, `run_test.{sh,bat}` gets run in addition to anything else you have in the `meta.yaml's` `test` block. (At least, it definitely ran with the previous PR, because it failed for my first attempt on cross-compile archs.)

Reply:
Yep you're right. Did a bit more digging and it does seem Conda-Build should handle this correctly ( https://github.com/conda-forge/conda-forge.github.io/issues/1490#issuecomment-1097336276 )

Reply:
Ah sorry I misinterpreted and thought you already knew that 😂

Yeah they should work together, but as noted in the issue there may have been edge cases where that wasn't true.

Should add normally the commands are echoed (as in [this CI build]( https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=489010&view=logs&j=656edd35-690f-5c53-9ba3-09c10d0bea97&t=e5c8ab1d-8ff9-5cae-b332-e15ae582ed2d&l=17342 )), but it looks like that output is suppressed when adding them to the end of `run_test.*` scripts. Maybe this is leading to confusion about whether they are being run?

Added some options to the `run_test.*` scripts, which should hopefully make it clearer that those tests are being run.

Perhaps part of the answer is Conda-Build should run `run_test.*` with these verbose options included?

Reply:
Hmm...so I do see the compiler test running [here]( https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=491401&view=logs&j=656edd35-690f-5c53-9ba3-09c10d0bea97&t=e5c8ab1d-8ff9-5cae-b332-e15ae582ed2d&l=17405 ). Though am not seeing the file tests afterwards. Maybe I'm still missing something?

Reply:
Ok think I figured out what is going on here. Filed issue ( https://github.com/conda/conda-build/issues/4427 ).